scripts/training/training_utils.py

```
@@ -0,0 +1,232 @@
+                "available": True,
+                "count": torch.cuda.device_count(),
+                "current": torch.cuda.current_device(),
+                "name": torch.cuda.get_device_name(0),
+                "memory_total": torch.cuda.get_device_properties(0).total_memory / 1e9
+            }
+        return {"available": False}
+    except ImportError:
+        return {"available": False, "error": "PyTorch not available"}
+
+
+def validate_training_data(
```

Ruff (0.12.2)

Too many return statements (8 > 6)

(PLR0911)